lib/src/push.dart

```diff
@@ -81,8 +90,7 @@ class Push {
         _responseCompleter = Completer<PushResponse>();
 
   final Logger _logger;
-  final ListMultimap<String, void Function(PushResponse)> _receivers =
-      ListMultimap();
+  final Map<String, List<ReceiverCallback>> _receivers = {};
```

Comment:
I'm less onboard with this change. Manipulating this map of list is done potentially very frequently, and doing it efficiently and without logic errors is critical. 

So I'd explore this change separately. Besides, quiver is a library that changes very infrequently and has a strong focus on API stability.

Reply:
If your concern is about performances of the `onReply` method, I've benchmarked the new implementation a bit and it was in fact slower than the one with `ListMultimap`.
**BUT**
I've also tested with my previous implementation:
```dart
// Note that this can actually be a 1-liner, re-assigning _receivers[status] is not needed.
(_receivers[status] ??= [])..add(callback)
```
And I've had **better performance** than with `ListMultimap`.

## Example

For **5000 map entries**, each containing a **list of 1000 elements**:
```
Start measuring ListMultimap performance for 5000 iterations (1000 items)
Quiver performance: 0:00:00.324885
Start measuring Map performance for 5000 iterations (1000 items)
Map performance: 0:00:00.240118
Map is faster by 0:00:00.084767
```

<details>
  <summary>Source code</summary>
  
  ```dart
  import 'package:quiver/collection.dart';
  
  const _kIterations = 5000;
  const _kListSize = 1000;
  
  typedef ReceiverCallback = void Function(PushResponse response);
  
  Duration quiverPerf() {
    final map = ListMultimap<String, ReceiverCallback>();
    final stopwatch = Stopwatch()..start();
    for (int i = 0; i < _kIterations; i++) {
      for (int j = 0; j < _kListSize; j++) {
        map['$i'].add((_) {});
      }
    }
    stopwatch.stop();
    return stopwatch.elapsed;
  }

  Duration mapPerf() {
    final map = <String, List<ReceiverCallback>>{};
    final stopwatch = Stopwatch()..start();
    for (int i = 0; i < _kIterations; i++) {
      for (int j = 0; j < _kListSize; j++) {
        (map['$i'] ??= []).add((_) {});
      }
    }
    stopwatch.stop();
    return stopwatch.elapsed;
  }

  class PushResponse {
    final String message;
    final int code;

    const PushResponse(this.message, this.code);
  }
  ```
</details>

Reply:
In your benchmark, I'd be curious to include in your cycle the cleaning part (removing items and removing the key when the list becomes empty).

Reply:
Are you talking about calls to `clearReceivers`? If yes, I've tested by clearing the previously created iterable and the Map implementation is still ahead:
```
[Clearing] Quiver performance: 0:00:00.000711
[Clearing] Map performance: 0:00:00.000007
[Clearing] Map is faster by 0:00:00.000704
```

Reply:
Note that in the end those values are very close, both seem well optimized. In my opinion performance shouldn't be the deciding factor here if quiver should be kept or not. The reality is that you were only using the `ListMultimap` class from the package, which seems to me a bit overkill considering how easy it is to replace.